haiku-apps/heidi/heidi-1.0.recipe

```diff
@@ -11,24 +11,25 @@ CHECKSUM_SHA256="dca3b020f36e83d0dc83b10b1362b73c6f2240605122fefbfc554e49ad70df8
 SOURCE_FILENAME="Heidi-$portVersion-$srcGitRev.tar.gz"
 SOURCE_DIR="Heidi-$srcGitRev"
 
-ARCHITECTURES="x86_gcc2 x86_64"
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
 
 PROVIDES="
-	heidi= $portVersion
-	app:Heidi = $portVersion
+	heidi$secondaryArchSuffix = $portVersion
+	app:Heidi$secondaryArchSuffix = $portVersion
```

Comment:
* Suggestion: conditional `SECONDARY_ARCHITECTURES`
* Minor fix: For `app:` items (only) we tend to drop the `$secondaryArchSuffix`.
```diff
 ARCHITECTURES="!x86_gcc2 x86_64"
+if [ "$targetArchitecture" = x86_gcc2 ]; then
 SECONDARY_ARCHITECTURES="x86"
+fi
 
 PROVIDES="
-	app:Heidi$secondaryArchSuffix = $portVersion
+	app:Heidi = $portVersion
```

Reply:
I thought that we already established this conditional syntax should not be added.
Others think so too: https://github.com/haikuports/haikuporter/issues/66#issuecomment-432246031

Reply:
@fbrosson The syntax is useless, as I said. So please don't add it where it's unnecessary.

Reply:
I have already explained in many messages why we should be using this.

Reply:
Well, @KapiX has started a discussion on the HaikuPorts ML: https://www.freelists.org/post/haikuports/Vote-Conditional-syntax-around-SECONDARY-ARCHITECTURES

As I replied, I agree that the "performance improvement" is irrelevant and the creation of extra directories is really meaningless. If we want to solve this, it should be done in HaikuPorter.

Reply:
FWIW, I never spoke about "performance improvement".
What matters here is just correctness. And "fixing" haikuporter is not the correct answer because there is no 100% correct automatic method to decide whether or not x86 secondary arch should be enabled. There is a general rule, but it cannot cover all corner cases such as a recipe providing libs that are of no use to any other package.

Don't worry, I'm not going to fight. This is my last message.